### ISSUE TITLE
Getting 'Uncaught' when setting up offline

### ISSUE THREAD

**Issue by gcirs** [naptha/tesseract.js#241]
First of all, I'm on an Angular 7 project.

I downloaded the required files to run Tesseract.js offline and placed them inside assets like so:
![image](https://user-images.githubusercontent.com/43372700/48409616-9776cf00-e73c-11e8-931a-130107a5ee6e.png)

Inside my component I've defined a new `Tesseract` instance like so:
```typescript
    this.tesseract = Tesseract.create({
      corePath: '../assets/lib/tesseract.js/tesseract.js',
      workerPath: '../assets/lib/tesseract.js/worker.js',
      langPath: '../assets/lib/tesseract-lang/'
    });
```

And then I run it:
```typescript
    this.tesseract.recognize(webcamImage.imageAsDataUrl)
      .progress(message => console.log(message))
      .catch(err => console.error(err))
      .then(res => console.log(res))
      .finally(resultOrError => console.log(resultOrError));
```
Which returns:
![image](https://user-images.githubusercontent.com/43372700/48409723-f0defe00-e73c-11e8-89ce-f4ff7348cef0.png)

If I try to run it "online" without using my instance of Tesseract:
```typescript
    Tesseract.recognize(webcamImage.imageAsDataUrl)
      .progress(message => console.log(message))
      .catch(err => console.error(err))
      .then(res => console.log(res))
      .finally(resultOrError => console.log(resultOrError));
```
Works perfectly.

Any ideas? Thank you in advance.

**Comment by Riyaz0001** [comment]
same problem.

**Comment by pablopunk** [comment]
Same error for me:

* Using [parcel-bundler](https://github.com/parcel-bundler/parcel)
* Installed with npm
* Using the browser camera to write into a canvas
* Using the canvas context as an argument for `recognize`

**Comment by EvgenyKislyy** [comment]
Same here

**Comment by Undefined01** [comment]
I also meet the same situation, and I have solved it. But I don't know if you are meeting the same question as mine.

I used to using the following scripts and ruun my app through 'file://' protocol without a HTTP server:
```javascript
window.Tesseract = Tesseract.create({
  workerPath: './worker.min.js',
  langPath: './',
  corePath: './index.js',
})
...
```
And I got 'Uncaught' without any other description but '(anonymous) blob:file:///5191d18…b535-85e1a58f0244:1'.

I finally found [the line](https://github.com/naptha/tesseract.js/blob/master/dist/tesseract.js#L257) that throwed this error, and it seems that it is going to import worker.js.

I immediately thought of cross-domain issues, although in my impression the js file is allowed with cross-domain access.

So I tried to set up a HTTP server and run my app again.

It crashed again and this time I got 'blob:http://localhost/dad01fec-7264-44db-bbaf-ba820c9688af:1 Uncaught DOMException: Failed to execute 'importScripts' on 'WorkerGlobalScope': The URL './worker.min.js' is invalid.'

Finally I change the script like this
```javascript
window.Tesseract = Tesseract.create({
  workerPath: 'http://localhost/worker.min.js',
  langPath: 'http://localhost/',
  corePath: 'http://localhost/index.js',
})
```

And finally it works perfectly.

Although I don't know why it was behaving like this, I am sharing my experience and hope to provide you with a reference.

**Comment by bmdproject99** [comment]
After considerable banging-head-against-wall, I now have tesseract.js working purely offline, using Node and Electron. I cleared cache in my electron app, disconnected internet, rebuilt the app, and verified that it works purely offline.

Here's what I did:


1. I downloaded  tesseract.js   and  worker.js   from:
   https://github.com/naptha/tesseract.js/tree/master/dist

I downloaded the core index.js from:
https://cdn.jsdelivr.net/gh/naptha/tesseract.js-core@0.1.0/index.js

I downloaded the lang file I need (English) from:
https://github.com/tesseract-ocr/tessdata

I put all four files into /scripts


2. I edited  tesseract.js  as follows:

In the spawnWorker function, I commented out all lines with Blob, leaving only
var worker = new Worker(workerOptions.workerPath).

See below (comments are mine):

exports.spawnWorker = function spawnWorker(instance, workerOptions) {
// if (window.Blob && window.URL) {
// var blob = new Blob(['importScripts("' + workerOptions.workerPath + '");']);
// var worker = new Worker(window.URL.createObjectURL(blob));
// } else {
var worker = new Worker(workerOptions.workerPath);
// }

worker.onmessage = function (e) {
    var packet = e.data;
    instance._recv(packet);
};
return worker;
};

I suspect there is a CORS issue with Blob in Chomium. Commenting out the lines removed the "Uncaught Blob file" error I was getting. Kudos to the following for suggesting this:
[#241 (comment)](https://github.com/naptha/tesseract.js/issues/241#issuecomment-450621164)


3. I included the  script in my index.html page:

<script src='scripts/tesseract.js'></script>


4. Here is the correct way to set the paths for  workerPath, langPath,  and corePath

(Note: in my example, all files are in /scripts. Adjust path accordingly).

I included the following at the top of my script page:

window.Tesseract = Tesseract.create({
workerPath: './scripts/worker.js',
langPath: '',
corePath: 'index.js'
})

It turns out langPath and corePath are relative to worker.js!!!

I hope this helps.
